apps/aechannel/src/aesc_fsm.erl

```diff
@@ -2659,7 +2664,10 @@ on_chain_id(D, SignedTx) ->
     {PubKey, D#data{on_chain_id = PubKey}}.
 
 initialize_cache(#data{ on_chain_id = ChId
-                      , state       = State } = D) ->
+                      , state       = State } = D, {ok, StatePassword}) ->
```

Comment:
The API is rather un-intuitive. I suggest either to provide the password or the fun and evaluate the fun in this function instead.